src/gui/processing/qgsprocessingfieldmapwidgetwrapper.cpp

```diff
@@ -130,6 +131,27 @@ void QgsProcessingFieldMapPanelWidget::setValue( const QVariant &value )
                 map.value( QStringLiteral( "length" ), 0 ).toInt(),
                 map.value( QStringLiteral( "precision" ), 0 ).toInt() );
 
+    int layerFieldIdx = layerFields.indexFromName( f.name() );
+
+    if ( mLayer && layerFieldIdx >= 0 && ! map.contains( QStringLiteral( "constraints" ) ) )
+    {
+      f.setConstraints( layerFields.at( layerFieldIdx ).constraints() );
```

Comment:
_(This was going to be a new QGIS issue, but I know where the behavior comes from, so I prefer to comment it here.)_

I think we would not needed to set constraints from `mLayer`, because it could be the source/input layer.

Highlighting constraints is relevant only if they are obtained from the target/template layer.

-------------------------------------
**Example**

This two point tables share the field `exactitud_vertical`, but have different constraints set on it.

```
CREATE TABLE survey.lc_puntocontrol (
	t_id bigserial NOT NULL,
	tipo_punto_control int8 NULL,
	exactitud_horizontal numeric(5,3) NOT NULL,
	exactitud_vertical numeric(5,3) NOT NULL,
	CONSTRAINT lc_puntocontrol_pkey PRIMARY KEY (t_id)
);

CREATE TABLE survey.lc_puntolindero (
	t_id bigserial NOT NULL,
	puntotipo int8 NOT NULL,
	exactitud_horizontal numeric(5,3) NOT NULL,
	exactitud_vertical numeric(5,3) NULL,
	CONSTRAINT lc_puntolindero_pkey PRIMARY KEY (t_id)
);
```

Steps to reproduce:

1. Open the Refactor Fields algorithm.
2. Set the template layer first (as `lc_puntolindero`).
3. Now set the INPUT layer (as `lc_punto_control`).
4. You get:
![image](https://user-images.githubusercontent.com/652785/99419270-ec202800-28c9-11eb-9a83-62739e5e8fd4.png)

As you can see, `exactitud_vertical` is highlighted, but it shouldn't, since the template layer has no constraint on such field.

-------------------------------------

Could we perhaps remove the `if ()` part above and only leave the `else` part (with its corresponding condition, of course)?

@suricactus @m-kuhn

Reply:
Yes, of course. Thank you for the detailed explanation what is expected! I will address this issue to match the expected behaviour.

Reply:
hey, @gacarrillor , sorry for the delay. This should fix the issue: https://github.com/qgis/QGIS/pull/40703

Reply:
Thank you @suricactus :+1: